content/en/docs/contribution-guidelines/upgrade-graphs.md

```diff
@@ -0,0 +1,241 @@
+
+```js
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+```
+
+Following the code semantic with examples which ought to be used to create the Upgrade Graphs Diagrams. 
+
+| Description   |      Code      |  Examples |
+|----------|:-------------:|------:|
+| head bundle |  `ID(<bundle tag>):::head` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1):::head
+{{</mermaid>}} |
+| installed bundle |  `ID(<bundle tag>):::installed` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1):::installed
+{{</mermaid>}} |
+| installable bundle | `ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1)
+{{</mermaid>}} |
+| channel |  `subgraph <channel name> end` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   subgraph preview
+      ID(v0.0.1)
+   end
+{{</mermaid>}} |
+| default channel | `subgraph "<channel name> (default)" end` |{{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   subgraph "preview (default)"
+      ID(v0.0.1) --> ID2(v0.0.2)
+   end
+{{</mermaid>}} |
+| replaces |`ID(<bundle tag>) --> ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) --> B(v0.0.4)
+{{</mermaid>}} |
+| skips | `ID(<bundle tag>) x--x | <versions that should be skipped> | ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) x--x |v0.0.2,v0.0.3| B(v0.0.4)
+{{</mermaid>}} |
+| skipRange | `ID<bundle tag>) o--o | <range> | ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   C(v2.0.3) o--o |>= 2.0.4 < 3.0.0| D(v3.0.1)
+{{</mermaid>}} |
+| future replaces |`ID(<bundle tag>) -.-> ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) -.-> B(v0.0.4)
+{{</mermaid>}} |
+| future skips | `ID(<bundle tag>) x-.-x | <versions that should be skipped> | ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) x-.-x |v0.0.2,v0.0.3| B(v0.0.4)
+{{</mermaid>}} |
+| future skipRange | `ID<bundle tag>) o-.-o | <range> | ID(<bundle tag>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   C(v2.0.3) o-.-o |>= 2.0.4 < 3.0.0| D(v3.0.1)
+{{</mermaid>}} |
+| Index image label  |    `ID(<bundle tag> \n fa:fa-tag <vendor version range>)` | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   E(v0.0.2 \n fa:fa-tag ocp=v4.x)
```

Comment:
Can this `ocp` label be switched for something non-openshift related?

Reply:
I understand that this feature is valid for community operators as well. But I think it is just OSDK too. have you any suggestion?

Reply:
It's for downstream community operators though. I think we should be doing whatever we can to avoid putting any OCP specific content in our upstream documentation if possible.

Reply:
Community-operators also has this feature. However, I am not sure if other kind vendors are supported.